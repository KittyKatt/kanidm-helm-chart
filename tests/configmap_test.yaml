# yaml-language-server: $schema=https://raw.githubusercontent.com/helm-unittest/helm-unittest/main/schema/helm-testsuite.json
suite: Test Configmap
templates:
  - templates/configmap.yaml
chart:
  version: 0.1.0
  appVersion: 1.4.5
release:
  name: kanidm-helm-test
  namespace: test
tests:
  - it: should exist and set correct Kind
    asserts:
      - isKind:
          of: ConfigMap
      - hasDocuments:
          count: 1
      - containsDocument:
          kind: ConfigMap
          apiVersion: v1

  - it: should set correct namespace
    asserts:
      - equal:
          path: metadata.namespace
          value: test

  - it: should set correct chart details
    asserts:
      - matchRegex:
          path: metadata.name
          pattern: kanidm
      - equal:
          path: metadata.labels
          value:
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: kanidm
            app.kubernetes.io/instance: kanidm-helm-test
            app.kubernetes.io/version: 1.4.5
            app.kubernetes.io/component: config
            helm.sh/chart: kanidm-0.1.0

  - it: should include extra labels
    set:
      global.labels.app0: bippity
    asserts:
      - isSubset:
          path: metadata.labels
          content:
            app0: bippity

  - it: should contain server configuration file
    asserts:
      - exists:
          path: data["server.toml"]
      - isNotNullOrEmpty:
          path: data["server.toml"]

  - it: should set sane default configuration
    set:
      kanidm:
        domain: idm.example.com
        origin: web.idm.example.com
    asserts:
      - matchSnapshot:
          path: data["server.toml"]

  - it: should disable backups if disabled in user input
    set:
      kanidm:
        backups:
          enabled: false
    asserts:
      - notMatchRegex:
          path: data["server.toml"]
          pattern: \[online_backup\]
