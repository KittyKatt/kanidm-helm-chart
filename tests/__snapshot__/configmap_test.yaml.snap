should set sane default configuration:
  1: |
    |
      bindaddress = "[::]:8443"
      ldapbindaddress = "[::]:3636"
      trust_x_forward_for = true
      tls_chain = "/certs/tls.crt"
      tls_key = "/certs/tls.key"
      db_path = "/data/kanidm.db"
      domain = "idm.example.com"
      origin = "https://web.idm.example.com"
      [online_backup]
      path = "/data/backups"
      schedule = "00 00 * * *"
